### DISCUSSION TITLE
Web is not shown on mobile phones,and can we set up ssl?

### DISCUSSION THREAD

**Discussion by Xiaye2332** [Dantevg/WebStats#36] [General]
I used my own web server. But when I add the ssl , the table that shows player stats is gone,and  I can not use my cell phone to see the table (other things like the title,next page button is visible).I think there is something wrong with the javascript, but can you fix it or show some solutions?

**Comment by Dantevg** [comment]
Hi! You're the first one using the discussions tab, nice!

HTTPS is not supported without some manual set-up. This is caused by two things.
- At the moment the plugin does not support serving the stats over HTTPS.
- Browsers do not allow plain HTTP requests from webpages served over HTTPS.

The result of this is that the webpage (which you serve over HTTPS) is not allowed to request the insecure (non-HTTPS) stats from WebStats running on the Minecraft server.

You can set up a reverse proxy on your web server. The wiki has an example for Nginx: https://github.com/Dantevg/WebStats/wiki/FAQ#https. If your server uses Apache (.htaccess files), [the Apache docs](https://httpd.apache.org/docs/2.4/howto/reverse_proxy.html) may be a good resource (I have not tried so I don't know).

Support for HTTPS in the plugin (without reverse proxy) is something I want to have in the future (issue #23), but it will not come soon.

**Reply by Xiaye2332** [reply]
All right.So what about the problem that the stats table do not shown on phones?

**Reply by Dantevg** [reply]
Does it work on desktops but specifically not on phones? If you're using HTTPS without the reverse proxy it should not work anywhere.

**Reply by Xiaye2332** [reply]
I do use http,and is OK on desktops but show nothing on phones.

**Comment by Xiaye2332** [comment]
![image](https://user-images.githubusercontent.com/86192453/169847524-80f7c518-326e-4353-91d7-7c4abd8b46bb.png)
![BF3203361B9A08F222E3A202DC18D078](https://user-images.githubusercontent.com/86192453/169847588-56844d86-1a97-4417-8d38-9f9a2b57e350.png)

Yes,the problem is only the table is missing,so strange.

**Reply by Dantevg** [reply]
Hmm, it sounds like it could potentially be a caching issue. It's a shame we can't see the console output on mobile. Does it work if you add something to the end of the js file script tag url? For example the unix timestamp like `<script src="path/to/webstats.js?t=1653318577"></script>`

**Reply by Xiaye2332** [reply]
but actually I have cleared all cache on my phone,and do not work,the same as use  the unix timestamp.

**Reply by Dantevg** [reply]
Could you share the URL? Then I can see if it also does not work on my phone, and do some testing if it indeed also doesn't work.

**Reply by Xiaye2332** [reply]
Yes, The URL is  http://ddns.xiayeqz.com:2888

**Reply by Dantevg** [reply]
It works on my phone, using Firefox. Since you seem to be using safari, I fear it is a browser issue. Since I can't install safari on windows, it'll take some time to test and debug. In the meantime you can use a different browser on mobile (if that's possible on IOS?)

**Reply by Xiaye2332** [reply]
It do not work on Chrom/Firefox on iOS,and also Safari.I have tried that. Since I do not have Andriod device, maybe it's a problem with iOS?

**Reply by Xiaye2332** [reply]
On macOS, even in destop mode, the browser is the same situation as phone.(Not only safari)it might  be a problem I think.

**Reply by Dantevg** [reply]
Very interesting, thanks for the testing! I do have access to macOS, so I'll see what I can find. It'll still take some time to test and debug though, as I don't have a lot of free time at the moment.

**Reply by Xiaye2332** [reply]
Thanks, it's my pleasure to do this. Maybe you can use app like Xcode On macOS to simulate iOS?(just suggestion).Wish the plugin could be better!

**Reply by Xiaye2332** [reply]
Another issues(maybe) is that in javascript file line 244, which is `if(img) img.src = https://www.mc-heads.net/avatar/${entry}.png`.It's shame that the api is not working in my region(China), reposing 403 Forbidden.So I replace that with another api `if(img) img.src = https://minotar.net/avatar/${entry}.png` , and it works. Maybe it can be a solution to someone who can not use that api too.

**Reply by Dantevg** [reply]
Ah, thanks for noting that! minotar.net seems to be working just as well as mc-heads.net so I'll just switch entirely to minotar.net in the next update.

**Reply by Dantevg** [reply]
Update: I've found the cause of the issue. It's because safari does not yet support Javascript regex lookbehind (https://caniuse.com/js-regexp-lookbehind). For me it works fine in Firefox on macOS. The reason it doesn't work with other browsers on IOS is that they still always have to use safari's browser engine. Will see how I can fix it somewhere this week.

**Reply by Dantevg** [reply]
Finally had time again to look at this, sorry for taking so long. Here's the code that should work on safari (and using minotar.net). I'm not able to test it on safari today but will do that later this week if I have time.

[WebStats-dist.zip](https://github.com/Dantevg/WebStats/files/9001336/WebStats-dist.zip)

**Reply by Xiaye2332** [reply]
Its awesome! Thank you very much :)

**Reply by Xiaye2332** [reply]
I have tested it and it is great. You can see that on [URL](https://rank.xiayeqz.com) . Thanks for doing so much.

**Reply by Dantevg** [reply]
Nice! I'm very happy to hear and see that!

**Comment by Xiaye2332** [comment]
Of course. &nbsp;
zn.zamcraft.net:98

------------------ Original ------------------
From: RedPolygon ***@***.***&gt;
Date: Mon,May 23,2022 11:51 PM
To: Dantevg/WebStats ***@***.***&gt;
Cc: Xiaye2332 ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [Dantevg/WebStats] Web is not shown on mobile phones,and can we set up ssl? (Discussion #36)





 
Could you share the URL? Then I can see if it also does not work on my phone, and do some testing if it indeed also doesn't work.

**Reply by Dantevg** [reply]
Hi again! Are you having an issue again / what is the problem?

**Reply by Xiaye2332** [reply]
Nope, Thank you.The plugin is really fantastic.

**Reply by Dantevg** [reply]
Oh, ok. Great! 😄
